### PR TITLE
Merge AddViewRequestDeleteFriend Feature Branch

### PR DESCRIPTION
**Pull request by JoeZZG**
also edit some actions in add friend feature

### CODE REVIEW

**Review by xingjianll — APPROVED** [review]
Approved.

**Review by katerynablavt — APPROVED** [review]

File: src/main/java/usecases/friendinteractors/viewfriend/view_friend_interactor.java [hunk 1]
```diff
@@ -22,6 +22,12 @@ public void ViewFriend(view_friend_input_model model) {
         ArrayList<CommonUser> arraylist = new ArrayList<CommonUser> ();
         arraylist = dataBase.getFriendList(model.requesterid);
```
Comment (xingjianll) [inline src/main/java/usecases/friendinteractors/viewfriend/view_friend_interactor.java:23]: You can shorten this to:
 ArrayList<CommonUser> arraylist = dataBase.getFriendList(model.requesterid);

Also, it should be arrayList.

File: src/main/java/usecases/friendinteractors/viewfriend/view_friend_output_boundary.java [hunk 2]
```diff
@@ -10,6 +10,5 @@ public interface view_friend_output_boundary {
      * @param arraylist
      */
     void pushFriendList(ArrayList<CommonUser> arraylist);
```
Comment (xingjianll) [inline src/main/java/usecases/friendinteractors/viewfriend/view_friend_output_boundary.java:12]: arrayList not arraylist.

**Review by xingjianll — CHANGES_REQUESTED** [review]
I'm not sure if a viewFriend usecase is even necessary in the first place since when a client accesses its friends list, it will have access to the data locally, hence there is no need for a request to be made to the server and trigger a Usecase. Also, PLEASE fix the name for the files!

**Review by davidlyj — DISMISSED** [review]
I think the tests do not need to be commented out. Other than that they are okay.

File: src/main/java/usecases/friendinteractors/viewfriend/view_friend_output_boundary.java (hunk 2 above)
Comment (JoeZZG) [reply src/main/java/usecases/friendinteractors/viewfriend/view_friend_output_boundary.java:12]: Changed :)

File: src/main/java/usecases/friendinteractors/viewfriend/view_friend_interactor.java (hunk 1 above)
Comment (JoeZZG) [reply src/main/java/usecases/friendinteractors/viewfriend/view_friend_interactor.java:23]: Changed :)

**Review by xingjianll — DISMISSED** [review]
Nice!

**Review by katerynablavt — APPROVED** [review]

**Review by xingjianll — APPROVED** [review]
I will approve your code, but I really think that having an viewFriend usecase is unecessary, since when a user on the client view the list of friends from their local Model, they do not have to make a request to the server to fetch the data again. Hence I suggest you to discard this usecase and focus on the other two in the future.

**Review by brendenmcfarlane — APPROVED** [review]
Loogs good, idk why everything had to get deleted and retyped
